metricflow/inference/renderer/config_file.py

```diff
@@ -0,0 +1,154 @@
+    is_primary: NotRequired[bool]
+
+
+class RenderedColumnConfig(TypedDict):  # noqa: D
+    name: str
+    type: str
+    type_params: NotRequired[RenderedTimeColumnConfigTypeParams]
+
+
+class ConfigFileRenderer(InferenceRenderer):
+    """Writes inference results to a set config files"""
+
+    UNKNOWN_FIELD_VALUE = "FIXME"
+
+    def __init__(self, dir_path: Union[str, Path], overwrite: bool = False) -> None:
+        """Initializes the renderer.
+
+        dir_path: The path to the config directory
+        overwrite: If set to False, will raise error if the directory exists
+        """
+        dir_path = os.path.abspath(dir_path)
+
+        if not overwrite and os.path.exists(dir_path):
+            raise ValueError("ConfigFileRender.overwrite is False but path exists.")
+
+        if os.path.isfile(dir_path):
+            raise ValueError("ConfigFileRenderer `dir_path` is a file.")
+
+        self.dir_path = dir_path
+
+    def _get_filename_for_table(self, table: SqlTable) -> str:
+        return os.path.abspath(os.path.join(self.dir_path, f"{table.sql}.yaml"))
+
+    def _fixme(self, comment: str) -> str:
+        return f"FIXME: {comment}"
+
+    def _render_id_columns(self, results: List[InferenceResult]) -> List[CommentedMap]:
+        type_map = {
+            InferenceSignalType.ID.PRIMARY: "primary",
+            InferenceSignalType.ID.FOREIGN: "foreign",
+            InferenceSignalType.ID.UNIQUE: "unique",
+        }
+
+        rendered: List[RenderedColumnConfig] = [
+            {
+                "name": result.column.column_name,
+                "type": type_map.get(result.type_node, ConfigFileRenderer.UNKNOWN_FIELD_VALUE),
+            }
+            for result in results
+            if result.type_node.is_subtype_of(InferenceSignalType.ID.UNKNOWN)
+        ]
+
+        return rendered
+
+    def _render_dimension_columns(self, results: List[InferenceResult]) -> List[CommentedMap]:
+        type_map = {
+            InferenceSignalType.DIMENSION.TIME: "time",
+            InferenceSignalType.DIMENSION.PRIMARY_TIME: "time",
+            InferenceSignalType.DIMENSION.CATEGORICAL: "categorical",
+        }
+
+        rendered: List[CommentedMap] = []
+        for result in results:
+            if not result.type_node.is_subtype_of(InferenceSignalType.DIMENSION.UNKNOWN):
+                continue
+
+            result_data: CommentedMap = CommentedMap(
+                {
+                    "name": result.column.column_name,
+                    "type": type_map.get(result.type_node, ConfigFileRenderer.UNKNOWN_FIELD_VALUE),
+                }
+            )
+
+            if result_data["type"] == ConfigFileRenderer.UNKNOWN_FIELD_VALUE:
+                result_data.yaml_add_eol_comment(self._fixme("unknown field value"), "type")
+
+            if result.type_node.is_subtype_of(InferenceSignalType.DIMENSION.TIME):
+                type_params: CommentedMap = {"time_granularity": "day"}
```

Comment:
We might want to add a fixme_ comment to this in a future update since we don't know the true granularity. I'll open an issue to track inference validation and we can update that once we learn more about how this works in practice.